src/include/buffer.h

```diff
@@ -181,35 +210,31 @@ namespace buffer CEPH_BUFFER_API {
 
     void release();
 
-  public:
-    class iterator {
+    template<bool is_const>
+    class iterator_impl : details::buffer_iterator_impl<is_const> {
+      using parent = details::buffer_iterator_impl<is_const>;
+      using parent::pos;
+      using parent::end_ptr;
       const ptr *bp;     ///< parent ptr
       const char *start; ///< starting pointer into bp->c_str()
-      const char *pos;   ///< pointer into bp->c_str()
```

Comment:
@dillaman it's already broken =(

see http://lists.ceph.com/pipermail/ceph-users-ceph.com/2018-January/023766.html . probably we can have a proper libradospp in nautilus.

Reply:
Hmm -- so that should affect Mimic, correct? It's odd that the client upgrade tests haven't started failing since that's usually the first indication that the ABI was broken.

Reply:
makes sense... i will double check the pre-mimic symbols. if this breaks the backward compatibility i will update this PR to address it.

Reply:
seems the noexcept specifier does not change the symbol name. and librbd/librados in mimic is still backward compatible.

fixed.